File: Version0.2/Julia/Preprocessing/mkfaceblocks.jl
function mkfaceblocks(mf,bcm,ns::Int=4096)

i = 1;
nf = mf[i+1]-mf[i];
nmf,nb = mkelemblocks(nf,ns);
tm = vcat(mf[i] .+ nmf, bcm[i]*ones(Int,1,size(nmf,2)));
nm = tm;

for i = 2:length(mf)-1
    nf = mf[i+1]-mf[i];
    nmf,nb = mkelemblocks(nf,ns);
    tm = vcat(mf[i] .+ nmf, bcm[i]*ones(Int,1,size(nmf,2)));
    nm = [nm tm];    
end
nb = size(nm,2);

return nm,nb

end

File: Version0.2/Python/Preprocessing/mkfaceblocks.py
from numpy import *
from mkelemblocks import mkelemblocks

def mkfaceblocks(mf,bcm,ns):

    for i in range(0,mf.size-1):
        nf = mf[i+1]-mf[i]
        nmf,nb = mkelemblocks(nf,ns);
        tm = vstack((mf[i]+nmf,bcm[i]*ones((1,nmf.shape[1]))));
        if i == 0:
            nm = tm;
        else:
            nm = hstack((nm, tm));
    nb = nm.shape[1];

    return nm,nb
